02_knight_and_princess/xieyu_wang/main.cpp

```diff
@@ -18,6 +22,14 @@ class Vertex {
     str += "]}";
     return str;
   }
+  bool hasEdgeId(int id) const {
+    for (const auto eId : edgeIds) {
```

Comment:
It is better to use reference instead of making a copy
```suggestion
    for (const auto& eId : edgeIds) {
```